### PR TITLE
🌱 Update dependencies

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What this PR does / why we need it**:
Updates a bunch of dependencies, the most important of which is corefile-migration to support Kubernetes 1.22.

Couldn't do etcd or go-rpc because of #4745 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by randomvariable** [comment]
/hold

Have a few more to do.

File: test/go.mod [hunk 1]
```diff
@@ -7,8 +7,8 @@ replace sigs.k8s.io/cluster-api => ../
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/go-logr/logr v0.4.0
-	github.com/onsi/ginkgo v1.16.3
-	github.com/onsi/gomega v1.12.0
+	github.com/onsi/ginkgo v1.16.4
+	github.com/onsi/gomega v1.13.0
```
Comment (vincepri) [inline test/go.mod:11]: FYI, these come from controller-runtime. We should update them first?

Comment (randomvariable) [reply test/go.mod:11]: https://github.com/kubernetes-sigs/controller-runtime/pull/1550

**Comment by randomvariable** [comment]
Rebased

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**